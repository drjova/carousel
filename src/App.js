import React, { Component } from 'react';

import Slider from './containers/Slider/Slider';

export default class App extends Component {
  render() {
    return (
      <div id="main">
        <Slider />
      </div>
    );
  }
}
