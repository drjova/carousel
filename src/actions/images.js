import { IMAGES_ERROR, IMAGES_REQUEST, IMAGES_SUCCESS } from './actionTypes';
import API_URL from '../constants';

function requestImages() {
  return {
    type: IMAGES_REQUEST,
  };
}

function errorImages(payload) {
  return {
    type: IMAGES_ERROR,
    payload,
  };
}

function successImages(payload) {
  return {
    type: IMAGES_SUCCESS,
    payload,
  };
}

export default function fetchImages() {
  return async (dispatch, getState, http) => {
    dispatch(requestImages());
    try {
      const response = await http.get(API_URL);
      dispatch(successImages(response.data));
    } catch (error) {
      dispatch(errorImages(error.response && error.response.data));
    }
  };
}
