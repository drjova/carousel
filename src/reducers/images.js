import {
  IMAGES_ERROR,
  IMAGES_REQUEST,
  IMAGES_SUCCESS,
} from '../actions/actionTypes';

export const initialState = {
  loading: false,
  data: [],
  error: null,
};

const imagesReducers = (state = initialState, action) => {
  switch (action.type) {
    case IMAGES_REQUEST:
      return { ...state, loading: true };
    case IMAGES_SUCCESS:
      return { ...state, loading: false, data: action.payload };
    case IMAGES_ERROR:
      return { ...state, loading: false, error: action.payload };
    default:
      return state;
  }
};

export default imagesReducers;
