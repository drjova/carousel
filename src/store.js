import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';

export const thunkMiddleware = thunk.withExtraArgument(http);

const PROD_MIDDLEWARES = [
    thunkMiddleware,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
    if (process.env.NODE_ENV === 'production') {
      return applyMiddleware(...PROD_MIDDLEWARES);
    }
    return applyMiddleware(...DEV_MIDDLEWARES);
};

export const store = createStore(
    reducers,
    composeWithDevTools(withMiddlewares()),
);