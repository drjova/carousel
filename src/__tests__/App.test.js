import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';

import App from '../App';
import { initialState } from '../reducers/images';
import { getStore, getStoreWithState } from '../fixtures/store';

describe('render App', () => {
  it('renders initial state', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
