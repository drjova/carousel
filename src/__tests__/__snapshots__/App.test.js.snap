// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App renders initial state 1`] = `
<div
  id="main"
>
  <Connect(Slider) />
</div>
`;
